### PR TITLE
update: adding example on iterating through groups

### PR DESCRIPTION
**Pull request by daniCsorbaJB**

### CODE REVIEW

File: docs/topics/collection-grouping.md [hunk 1]
```diff
@@ -39,12 +39,37 @@ Namely, `Grouping` supports the following operations:
...
+
+To iterate through the groups created by the `groupBy()` function, you can use the [`forEach`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each.html) function on the resulting `Map`.
+This allows you to access each key and the list of elements associated with that key.
+
+Let's look at an example where we group the strings by their first letter, convert it to uppercase,
+and then iterate through each group to print the key and its associated values:
+
+```kotlin
+fun main() {
+    val numbers = listOf("one", "two", "three", "four", "five")
```
Comment (ilya-g) [inline docs/topics/collection-grouping.md:59]: missing //sampleStart //sampleEnd comments

File: docs/topics/collection-grouping.md [hunk 2]
```diff
@@ -39,12 +39,37 @@ Namely, `Grouping` supports the following operations:
...
     val numbers = listOf("one", "two", "three", "four", "five", "six")
     println(numbers.groupingBy { it.first() }.eachCount())
 //sampleEnd
 }
 ```
-{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
\ No newline at end of file
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
+
+To iterate through the groups created by the `groupBy()` function, you can use the [`forEach`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each.html) function on the resulting `Map`.
```
Comment (ilya-g) [inline docs/topics/collection-grouping.md:51]: `for` operator works as well: https://kotlinlang.org/docs/coding-conventions.html#loops

**Review by ilya-g — CHANGES_REQUESTED** [review]
Overall, I believe it's better to extend existing samples with the explicit types of the result and iteration example than to reiterate it in a new section.

File: docs/topics/collection-grouping.md (hunk 2 above)
Comment (daniCsorbaJB) [reply docs/topics/collection-grouping.md]: I added it 👍

File: docs/topics/collection-grouping.md (hunk 1 above)
Comment (daniCsorbaJB) [reply docs/topics/collection-grouping.md:50]: good point -  I restructured it 👍

File: docs/topics/collection-grouping.md [hunk 3]
```diff
@@ -30,21 +30,38 @@ right before the operation execution.
...
+  perform [fold and reduce](collection-aggregate.md#fold-and-reduce) operations on each group as a separate collection
+  and return the results.
 * [`aggregate()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/aggregate.html) applies a given operation
-   subsequently to all the elements in each group and returns the result.
-   This is the generic way to perform any operations on a `Grouping`. Use it to implement custom operations when fold or reduce are not enough.
+  subsequently to all the elements in each group and returns the result.
+  This is the generic way to perform any operations on a `Grouping`. Use it to implement custom operations when fold or reduce are not enough.
 
-```kotlin
+To iterate through the groups created by the `groupBy()` function, you can use the [`forEach`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each.html) or the `for` functions on the resulting `Map`.
```
[the remark this answers is not in the document]
Comment (daniCsorbaJB) [reply docs/topics/collection-grouping.md]: Good insight, thank you - 
I modified the examples and descriptions - added the iteration to both! 👍

**Review by ilya-g — APPROVED** [review]

File: docs/topics/collection-grouping.md [hunk 4]
```diff
@@ -2,22 +2,33 @@
...
+lambda function and returns a `Map`. In this map, each key is the lambda result, and the corresponding value is the `List`
 of elements on which this result is returned. This function can be used, for example, to group a list of `String`s by
 their first letter. 
 
 You can also call `groupBy()` with a second lambda argument – a value transformation function.
 In the result map of `groupBy()` with two lambdas, the keys produced by `keySelector` function are mapped to the results
 of the value transformation function instead of the original elements.
 
-```kotlin
+This example illustrates using the `for` operator to iterate through the groups created by the `groupBy()` function on the resulting `Map`:
```
Comment (AlejandraPedroza) [inline docs/topics/collection-grouping.md:13]: I would suggest listing the actions in the same order as they appear in the code snippet: grouping by first letter, transforming to uppercase, and using `for`.

File: docs/topics/collection-grouping.md [hunk 5]
```diff
@@ -2,22 +2,33 @@
...
 
+```kotlin
 fun main() {
 //sampleStart
     val numbers = listOf("one", "two", "three", "four", "five")
 
-    println(numbers.groupBy { it.first().uppercase() })
-    println(numbers.groupBy(keySelector = { it.first() }, valueTransform = { it.uppercase() }))
+    // Groups the strings by their first letter using groupBy
```
Comment (AlejandraPedroza) [inline docs/topics/collection-grouping.md:20]:
```suggestion
    // Groups the strings by their first letter using groupBy()
```

File: docs/topics/collection-grouping.md [hunk 6]
```diff
@@ -2,22 +2,33 @@
...
+    val groupedByFirstLetter = numbers.groupBy { it.first().uppercase() }
+    println(groupedByFirstLetter)
```
Comment (AlejandraPedroza) [inline docs/topics/collection-grouping.md:22]: From what I remember, we add a comment line with the output right after the print line for those that cant' run the code snippets.

File: docs/topics/collection-grouping.md [hunk 7]
```diff
@@ -30,21 +41,35 @@ right before the operation execution.
...
-    val numbers = listOf("one", "two", "three", "four", "five", "six")
-    println(numbers.groupingBy { it.first() }.eachCount())
+    val numbers = listOf("one", "two", "three", "four", "five")
+
+    // Groups the strings by their first letter using groupingBy() and counts the elements in each group
+    val grouped = numbers.groupingBy { it.first() }.eachCount()
+
+    // Iterates through each group and prints the key and its associated values
+    for ((key, count) in grouped) {
+        println("Key: $key, Count: $count")
```
Comment (AlejandraPedroza) [inline docs/topics/collection-grouping.md:68]: Same as before, we can add the actual output to the code snippet. You can see some examples here: https://kotlinlang.org/docs/kotlin-tour-hello-world.html#string-templates

**Review by AlejandraPedroza — COMMENTED** [review]
Nice improvement to the docs! I left some comments, please take a look.

File: docs/topics/collection-grouping.md (hunk 6 above)
Comment (daniCsorbaJB) [reply docs/topics/collection-grouping.md:22]: looked a bit too busy for me with the output included as well, but sure - it cannot hurt 👍

File: docs/topics/collection-grouping.md [hunk 8]
```diff
@@ -2,22 +2,33 @@
...
+    // Groups the strings by their first letter using groupBy
+    val groupedByFirstLetter = numbers.groupBy { it.first().uppercase() }
+    println(groupedByFirstLetter)
+
+    // Groups the strings by their first letter and transforms the values to uppercase
+    val groupedAndTransformed = numbers.groupBy(keySelector = { it.first() }, valueTransform = { it.uppercase() })
+    println(groupedAndTransformed)
+
+    // Iterates through each group and prints the key and its associated values
+    for ((key, value) in groupedByFirstLetter) {
```
[the remark this answers is not in the document]
Comment (daniCsorbaJB) [reply docs/topics/collection-grouping.md:29]: hmm 🤔 - that's an excellent point. I presented it in the order of importance (keySelector is more imporant than the iteration). 

But I think it looks better implementing your suggestion - thank you! 🙏

**Review by AlejandraPedroza — APPROVED** [review]
LGTM 👍 🚀

**Pull request merged**